### PR TITLE
Cleanup FXIOS-3966 [v103] Move some web server logic out of the app delegate

### PR DESCRIPTION
**Pull request by OrlaM**
More app delegate clean up

### CODE REVIEW

**Comment by OrlaM** [comment]
This has passed on BR, not sure why it's showing a failure.
image

File: Client/Application/WebServer.swift [hunk 1]
```diff
@@ -6,7 +6,12 @@ import Foundation
 import GCDWebServers
 import Shared
 
-class WebServer {
+protocol WebServerProtocol {
+    var server: GCDWebServer { get }
+    @discardableResult func start() throws -> Bool
+}
+
+class WebServer: WebServerProtocol {
```
Comment (nishant2718) [inline Client/Application/WebServer.swift:14]: 👌

File: Client/Application/WebServerUtil.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        if AppConstants.IsRunningTest || AppConstants.IsRunningPerfTest {
+            registerHandlersForTestMethods(server: webServer.server)
+        }
+
+        // Bug 1223009 was an issue whereby CGDWebserver crashed when moving to a background task
+        // catching and handling the error seemed to fix things, but we're not sure why.
+        // Either way, not implicitly unwrapping a try is not a great way of doing things
+        // so this is better anyway.
+        do {
+            try webServer.start()
```
Comment (nishant2718) [inline Client/Application/WebServerUtil.swift:46]: Just a thought - not a PR blocker at all.

I know you didn't author it, but is this something we should be more concerned about? `start` is discardable and wrapped in a try catch without documentation explaining why a log statement is enough to continue. I guess we can investigate the consequence further in a future ticket (and maybe if we can defer this to a later point in app init!)

**Review by nishant2718 — APPROVED** [review]
Looks good!

**Pull request merged**